packages/next/client/future/image.tsx

```diff
@@ -380,34 +412,71 @@ export default function Image({
-        )
-      }
-      if (!VALID_LOADING_VALUES.includes(loading)) {
-        throw new Error(
-          `Image with src "${src}" has invalid "loading" property. Provided "${loading}" should be one of ${VALID_LOADING_VALUES.map(
-            String
-          ).join(',')}.`
-        )
+      if (fill) {
+        if (width) {
+          throw new Error(
+            `Image with src "${src}" has both "width" and "fill" properties. Images in fill mode should not specify width or height.`
+          )
+        }
+        if (height) {
+          throw new Error(
+            `Image with src "${src}" has both "height" and "fill" properties. Images in fill mode should not specify height or width.`
+          )
+        }
+        if (style?.position) {
+          throw new Error(
+            `Image with src "${src}" has a style attribute which modifies the CSS 'position' attribute. Fill images have position: 'absolute' and this should not be modified.`
+          )
+        }
+      } else {
+        if (typeof widthInt === 'undefined') {
+          throw new Error(
+            `Image with src "${src}" is missing required "width" property.`
+          )
+        } else if (isNaN(widthInt)) {
+          throw new Error(
+            `Image with src "${src}" has invalid "width" property. Expected a numeric value in pixels but received "${width}".`
+          )
+        }
+        if (typeof heightInt === 'undefined') {
+          throw new Error(
+            `Image with src "${src}" is missing required "height" property.`
+          )
+        } else if (isNaN(heightInt)) {
+          throw new Error(
+            `Image with src "${src}" has invalid "height" property. Expected a numeric value in pixels but received "${height}".`
+          )
+        }
       }
-      if (priority && loading === 'lazy') {
-        throw new Error(
-          `Image with src "${src}" has both "priority" and "loading='lazy'" properties. Only one should be used.`
+    }
+    if (!VALID_LOADING_VALUES.includes(loading)) {
+      throw new Error(
+        `Image with src "${src}" has invalid "loading" property. Provided "${loading}" should be one of ${VALID_LOADING_VALUES.map(
+          String
+        ).join(',')}.`
+      )
+    }
+    if (priority && loading === 'lazy') {
+      throw new Error(
+        `Image with src "${src}" has both "priority" and "loading='lazy'" properties. Only one should be used.`
+      )
+    }
+
+    if ('objectFit' in rest) {
+      throw new Error(
+        `Image with src "${src}" has unknown prop "objectFit". This style should be specified using the "style" attribute.`
+      )
+    }
+    if ('objectPosition' in rest) {
+      throw new Error(
+        `Image with src "${src}" has unknown prop "objectPosition". This style should be specified using the "style" attribute.`
+      )
+    }
+
+    if (placeholder === 'blur') {
```

Comment:
I think a few of these `if` statements somehow repeated during the merge with the canary branch so they can be removed now

Reply:
fixed